sklearn/cluster/tests/test_mean_shift.py

```diff
@@ -39,27 +38,34 @@ def test_estimate_bandwidth():
     assert 0.9 <= bandwidth <= 1.5
 
 
-def test_estimate_bandwidth_1sample():
+def test_estimate_bandwidth_1sample(global_dtype):
     # Test estimate_bandwidth when n_samples=1 and quantile<1, so that
     # n_neighbors is set to 1.
-    bandwidth = estimate_bandwidth(X, n_samples=1, quantile=0.3)
-    assert bandwidth == pytest.approx(0.0, abs=1e-5)
+    bandwidth = estimate_bandwidth(
+        X.astype(global_dtype, copy=False), n_samples=1, quantile=0.3
+    )
+
+    assert bandwidth.dtype == X.dtype
+    assert_allclose(bandwidth, 0.0, atol=1e-5)
 
 
 @pytest.mark.parametrize(
     "bandwidth, cluster_all, expected, first_cluster_label",
     [(1.2, True, 3, 0), (1.2, False, 4, -1)],
 )
-def test_mean_shift(bandwidth, cluster_all, expected, first_cluster_label):
+def test_mean_shift(
+    global_dtype, bandwidth, cluster_all, expected, first_cluster_label
+):
     # Test MeanShift algorithm
+    X_ = X.astype(global_dtype, copy=False)
     ms = MeanShift(bandwidth=bandwidth, cluster_all=cluster_all)
```

Comment:
I think that it could be nice to add a check for the `dtype` of the fitted attribute `ms.cluster_centers_`

Reply:
done below